firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceScreenTracesTest.java

```diff
@@ -37,24 +36,20 @@ public class FirebasePerformanceScreenTracesTest {
   private static final int LAUNCH_TIMEOUT = 5000;
 
   @Rule
-  public ActivityTestRule<FirebasePerfScreenTracesActivity> activityRule =
-      new ActivityTestRule<>(
-          FirebasePerfScreenTracesActivity.class,
-          /* initialTouchMode= */ false,
-          /* launchActivity= */ true);
+  public ActivityScenarioRule<FirebasePerfScreenTracesActivity> activityRule =
+      new ActivityScenarioRule<>(FirebasePerfScreenTracesActivity.class);
 
   @Test
   public void scrollRecyclerViewToEnd() {
-    RecyclerView recyclerView = activityRule.getActivity().findViewById(R.id.rv_numbers);
-    int itemCount = recyclerView.getAdapter().getItemCount();
+    ActivityScenario scenario = activityRule.getScenario();
+    int itemCount = FirebasePerfScreenTracesActivity.NUM_LIST_ITEMS;
```

Comment:
Isn't getting the count by referring to the recycler view a better idea? With a constant, we would not be able to adjust to any other changes to the list view later. Eg: we might be adding a separate list view item, but the scroll action here would ignore that since it thinks there are only NUM_LIST_ITEMS in the recycler view.

I would recommend reading it from the recycler view rather than a constant number.

Reply:
new `ActivityScenario` api doesn't have something like `getActivity()` so the old way is no longer working.